### PR TITLE
todo-review

### PR DESCRIPTION
**Pull request by MiracleHorizon**
## Summary by CodeRabbit

- **New Features**
    - Introduced dynamic loading for components in `ExtendDirectionForm` to improve performance.
    - Added a new skeleton component for loading placeholders in the `Extend Direction` section.
- **Enhancements**
    - Updated file naming functionality to include image dimensions for resized images.
    - Adjusted import paths and module structure for better organization and maintainability.
    - Enhanced file type checks for image resizing operations.
- **Refactor**
    - Reorganized settings and store-related code for clarity and efficiency.
- **Tests**
    - Added tests for the new `addImageSizesToFileName` functionality.

### CODE REVIEW

File: apps/frontend/src/lib/helpers/file/addImageSizesToFileName.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * @param width - image width in pixels
+ * @param height - image height in pixels
+ * @returns file name with image sizes added (example: 'logo-100x100.png')
+ */
+export function addImageSizesToFileName({ fullFileName, width, height }: Parameters): string {
+  const fileName = cropFileName(fullFileName)
+  const fileNameWithSizes = `${fileName}-${width}x${height}`
+  const fileExtension = fullFileName.replace(fileName, '')
+
+  return fileNameWithSizes + fileExtension
```
[the remark this answers is not in the document]
Comment (MiracleHorizon) [reply apps/frontend/src/lib/helpers/file/addImageSizesToFileName.ts:14]: Yea

[the remark this answers is not in the document]
Comment (MiracleHorizon) [reply apps/frontend/src/lib/helpers/file/addImageSizesToFileName.ts:14]: > The `addImageSizesToFileName` function correctly appends image dimensions to file names. However, consider handling edge cases where the `fullFileName` might not contain an extension or where `cropFileName` might return an unexpected format. Adding unit tests for this function would help ensure its reliability across various input scenarios.
> 
> Would you like me to help with creating unit tests for this function?

Yes
